base/src/main/java/com/mindera/skeletoid/dialogs/AbstractDialogFragment.kt

```diff
@@ -0,0 +1,181 @@
+    }
+
+    protected abstract fun disposeRxBindings()
+
+    fun setTargetActivity(requestCode: Int) {
+        targetActivityRequestCode = requestCode
+    }
+
+    override fun show(fragmentManager: FragmentManager?, tag: String) {
+
+        if (fragmentManager == null) {
+            LOG.e(LOG_TAG, Exception("Check StackTrace -> "),
+                    "Fragment.show():: FragmentManager cannot be null")
+            return
+        }
+
+        val activity = targetFragment?.activity ?: activity
+
+        if (isActivityFinishingOrNull(activity)) {
+            LOG.e(LOG_TAG, Exception("Invalid state for Activity"),
+                    "show(): Fragment Activity cannot be finishing or null...")
+            return
+        }
+
+        // If true allows only one with this tag to avoid multiple dialogs
+        if (isSingleTop && fragmentManager.findFragmentByTag(tag) is AbstractDialogFragment) {
+            LOG.e(LOG_TAG, "show(): Dialog already present for $tag")
+            return
+        }
+
+        val ft = fragmentManager.beginTransaction()
+        ft.add(this, tag)
+
+        LOG.d(LOG_TAG, "Committing Dialog transaction ", tag, " for dialog ", this.toString())
+
+        try {
+            ft.commit()
+        } catch (t: Throwable) {
+            LOG.e(LOG_TAG, "[Dialog] Failed to show: $tag")
+        }
+    }
+
+    private fun isActivityFinishingOrNull(activity: FragmentActivity?): Boolean {
+        return activity?.isFinishing ?: true
+    }
+
+    override fun onDismiss(dialog: DialogInterface?) {
+        super.onDismiss(dialog)
+        onDismiss()
+    }
+
+    override fun onCancel(dialog: DialogInterface) {
+        super.onCancel(dialog)
+        onCancel()
+    }
+
+    /**
+     * Dismiss the dialog normally, but if the state of the activity had already been saved dismiss
+     * dialog allowing state loss.
+     */
+    override fun dismiss() = try {
+        LOG.d("Dismissing dialog ", tag)
+        onDismiss()
+        super.dismiss()
+    } catch (ex: IllegalStateException) {
+        LOG.e(LOG_TAG, ex, "Dismissing dialog $tag allowing state loss. Activity = $activity isFinishing = ${activity?.isFinishing
+                ?: "null"}")
+        super.dismissAllowingStateLoss()
+    }
+
+    private fun passEventToTargetActivity(state: DialogState): Boolean {
+        if (activity is DialogFragmentHandler) {
+            if (targetActivityRequestCode == 0) {
+                LOG.e(LOG_TAG, "Invalid request code for activity")
+            } else {
+                (activity as DialogFragmentHandler).onDialogResult(targetActivityRequestCode, state, getContentIntent())
+                return true
+            }
+        }
+
+        return false
+    }
+
+    protected fun passEventToTargetFragment(state: DialogState): Boolean {
+        if (targetFragment is DialogFragmentHandler) {
+            (targetFragment as DialogFragmentHandler).onDialogResult(targetRequestCode, state, getContentIntent())
```

Comment:
does it still need the cast since `targetFragment` is a `val` it doesn't infer the type through the `if` condition?

Reply:
It really needs it:

https://cl.ly/d693935a0ca2